sklearn/utils/metaestimators.py

```diff
@@ -14,16 +14,22 @@ class _IffHasAttrDescriptor(object):
     """Implements a conditional property using the descriptor protocol.
 
     Using this class to create a decorator will raise an ``AttributeError``
-    if the ``attribute_name`` is not present on the base object.
+    if none of the delegates (specified in ``delegate_names``) is an attribute
+    of the base object or none of the delegates has an attribute
+    ``attribute_name``.
 
-    This allows ducktyping of the decorated method based on ``attribute_name``.
+    This allows ducktyping of the decorated method based on
+    ``delegate.attribute_name`` where ``delegate`` is the first item in
+    ``delegate_names`` that is an attribute of the base object.
```

Comment:
How about the following:

Using this class to create a decorator for a class method ``object.attribute_name``.
An ``AttributeError`` will be raised if none of the delegates (specified in
``delegate_names``) is an attribute of ``object`` (i.e. ``hasattr(object, delegate )``
is ``False``) or none of the delegates has an attribute ``attribute_name`` (i.e.
``hasattr(delegate, attribute_name)`` is ``False``).

This allows ducktyping of the decorated method ``object.attribute_name`` based on
``object.delegate.attribute_name`` where ``delegate`` is the first item in
``delegate_names`` for which ``hasattr(object, delegate )`` is ``True``).